### PR TITLE
helm: update docs to clarify chart maintenance requirements

### PR DESCRIPTION
**Pull request by jackfrancis**
#### What this PR does / why we need it:

This PR adds more detail to the helm chart maintenance documentation, especially how to manage chart version updates.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by jackfrancis** [comment]
FYI this go playground link proves that the recommended suffix matter is valid semver v2 (a Helm version requirement):

https://go.dev/play/p/2z9UZNXdOih

File: helm/cloud-provider-azure/README.md [hunk 1]
```diff
@@ -145,7 +145,42 @@ From the git root:
 
 ```bash
 $ make update-helm
-Successfully packaged chart and saved it to: helm/repo/cloud-provider-azure-1.23.8.tgz
+Successfully packaged chart and saved it to: helm/repo/cloud-provider-azure-1.23.11.tgz
 ```
 
 Changes to the chart should *always* include a new version, and then an update to the helm repo as described above.
+
+If changes to the chart don't include an update to the helm repo as described above, then the PR CI job will fail, preventing a merge to `master`.
```
Comment (mboersma) [inline helm/cloud-provider-azure/README.md:153]:
```suggestion
If changes to the chart don't include an update to the Helm repo as described above, then the PR CI job will fail, preventing a merge to `master`.
```

File: helm/cloud-provider-azure/README.md [hunk 2]
```diff
@@ -145,7 +145,42 @@ From the git root:
...
++ mkdir hack/../chart_verify
++ tar -xf hack/../helm/repo/cloud-provider-azure-1.23.11.tgz -C chart_verify
++ diff -r hack/../chart_verify/cloud-provider-azure/ hack/../helm/cloud-provider-azure/ --exclude README.md
++ rm -Rf hack/../chart_verify
++ exit 0
+$ echo $?
+0
+```
+
+## How to pick an appropriate new version when updating the helm chart
```
Comment (mboersma) [inline helm/cloud-provider-azure/README.md:180]:
```suggestion
## How to pick an appropriate new version when updating the Helm chart
```

File: helm/cloud-provider-azure/README.md [hunk 3]
```diff
@@ -145,7 +145,42 @@ From the git root:
...
+
+By convention, we pin the helm chart version to the most recent release of cloud-provider-azure supporting the most recent release of Kubernetes. For example, at the time of this writing, the most recent version of Kubernetes supported by cloud-provider-azure is 1.23, and the most recent release of cloud-provider-azure is `v1.23.11`. When the next version of cloud-provider-azure is released (either `v1.23.12`, or `v1.24.0` in the event that the next release includes support for a new minor release of Kubernetes) we would update the helm chart to refer to the newly released, versioned image references of `cloud-controller-manager` and `cloud-node-manager`, and then update the helm chart version to the same version of of the new release (e.g., either `v1.23.12` or `1.24.0`).
```
Comment (mboersma) [inline helm/cloud-provider-azure/README.md:182]:
```suggestion
By convention, we pin the Helm chart version to the most recent release of cloud-provider-azure supporting the most recent release of Kubernetes. For example, at the time of this writing, the most recent version of Kubernetes supported by cloud-provider-azure is 1.23, and the most recent release of cloud-provider-azure is `v1.23.11`. When the next version of cloud-provider-azure is released (either `v1.23.12`, or `v1.24.0` in the event that the next release includes support for a new minor release of Kubernetes) we would update the Helm chart to refer to the newly released, versioned image references of `cloud-controller-manager` and `cloud-node-manager`, and then update the Helm chart version to the same version of of the new release (e.g., either `v1.23.12` or `1.24.0`).
```

File: helm/cloud-provider-azure/README.md [hunk 4]
```diff
@@ -145,7 +145,42 @@ From the git root:
...
+
+If, however, changes to the chart are made independent of a new release of cloud-provider-azure, we can follow the semver convention of adding `suffix` data to the exiting release. For example, if the current release of the helm chart is `v1.23.11` and we want to add additional configuration capabilities to `values.yaml`, we may do so, and then update the version of the chart to `v1.23.11+20220505` following the formula `<existing release>+<current date in format YYYYMMDD>`.
```
Comment (mboersma) [inline helm/cloud-provider-azure/README.md:184]:
```suggestion
If, however, changes to the chart are made independently of a new release of cloud-provider-azure, we can follow the semantic versioning convention of adding `suffix` data to the exiting release. For example, if the current release of the Helm chart is `v1.23.11` and we want to add additional configuration capabilities to `values.yaml`, we may do so, and then update the version of the chart to `v1.23.11+20220505` following the formula `<existing release>+<current date in format YYYYMMDD>`.
```

File: helm/cloud-provider-azure/README.md [hunk 5]
```diff
@@ -145,7 +145,42 @@ From the git root:
...
+
+As described [above](#maintaining-the-repo), always run `make update-helm` after making changes to the helm chart, and updating the version.
```
Comment (mboersma) [inline helm/cloud-provider-azure/README.md:186]:
```suggestion
As described [above](#maintaining-the-repo), always run `make update-helm` after making changes to the Helm chart, and updating the version.
```

**Comment by jackfrancis** [comment]
@mboersma thanks! incorporated your suggestions

**Review by mboersma — APPROVED** [review]
/lgtm

File: helm/cloud-provider-azure/README.md [hunk 6]
```diff
@@ -139,13 +139,48 @@ The following configuration is made available for advanced users. There are no d
...
-Changes to the chart should *always* include a new version, and then an update to the helm repo as described above.
+Changes to the chart should *always* include a new version, and then an update to the Helm repo as described above.
+
+If changes to the chart don't include an update to the Helm repo as described above, then the PR CI job will fail, preventing a merge to `master`.
+
+To check if your Helm chart changes will pass CI, you can manually run the verification script:
+
+```bash
+$ hack/verify-helm-repo.sh
+++ dirname hack/verify-helm-repo.sh
```
Comment (nilo19) [inline helm/cloud-provider-azure/README.md:159]: is it possible to move the script into a file?

Comment (lzhecheng) [reply helm/cloud-provider-azure/README.md:159]: @nilo19 this is actually not a script. Just run the command `hack/verify-helm-repo.sh` and its output.

Comment (lzhecheng) [reply helm/cloud-provider-azure/README.md:159]: @jackfrancis I think you ran the old script and could you please rerun with the latest one? I just updated it yesterday.
https://github.com/kubernetes-sigs/cloud-provider-azure/pull/1656#event-6556909503

File: helm/cloud-provider-azure/README.md [hunk 7]
```diff
@@ -139,13 +139,48 @@ The following configuration is made available for advanced users. There are no d
...
+
+If, however, changes to the chart are made independently of a new release of cloud-provider-azure, we can follow the semantic versioning convention of adding `suffix` data to the exiting release. For example, if the current release of the Helm chart is `v1.23.11` and we want to add additional configuration capabilities to `values.yaml`, we may do so, and then update the version of the chart to `v1.23.11+20220505` following the formula `<existing release>+<current date in format YYYYMMDD>`.
```
Comment (lzhecheng) [inline helm/cloud-provider-azure/README.md:184]:
```suggestion
If, however, changes to the chart are made independently of a new release of cloud-provider-azure, we can follow the semantic versioning convention of adding `suffix` data to the existing release. For example, if the current release of the Helm chart is `v1.23.11` and we want to add additional configuration capabilities to `values.yaml`, we may do so, and then update the version of the chart to `v1.23.11+20220505` following the formula `<existing release>+<current date in format YYYYMMDD>`.
```

Comment (jackfrancis) [reply helm/cloud-provider-azure/README.md]: thanks!

File: helm/cloud-provider-azure/README.md (hunk 6 above)
Comment (jackfrancis) [reply helm/cloud-provider-azure/README.md:159]: updated output rebasing on top of your script change, thanks!

**Pull request merged**